### PR TITLE
unary scalar input processing

### PR DESCRIPTION
**Pull request by scxiao**
Unary returns a non-scalar shape for input scalar input shape, which does not make sense.

### CODE REVIEW

File: src/shape.cpp [hunk 1]
```diff
@@ -258,6 +258,10 @@ bool shape::transposed() const
 bool shape::broadcasted() const
 {
     assert(this->lens().size() == this->strides().size());
+    // a scalar is not considered as broadcasted.
+    if(this->lens().size() == 1 and this->lens()[0] == 1 and this->strides()[0] == 0)
```
Comment (pfultz2) [inline src/shape.cpp:262]: Scalar should be considered broadcast.

**Comment by pfultz2** [comment]
What is the reason for making this change?

**Comment by scxiao** [comment]
> What is the reason for making this change?

The unary ops changes a scalar input to a shape as {1}, {1}, this causes a shape mismatch of the cond parameter between the input/output of the submodule for the loop operator.

File: src/shape.cpp (hunk 1 above)
Comment (scxiao) [reply src/shape.cpp]: OK, I removed this change.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**